CoreDistributedCache/NHibernate.Caches.CoreDistributedCache/CoreDistributedCache.cs

```diff
@@ -175,24 +174,18 @@ private static bool GetUseSlidingExpiration(IDictionary<string, string> props)
 
 		private string GetCacheKey(object key)
 		{
+			var baseKey = _cacheKeyPrefix + _fullRegion + ":" + key;
 			var keyAsString = AppendHashcodeToKey
-				? string.Concat(_cacheKeyPrefix, _fullRegion, ":", key.ToString(), "@", key.GetHashCode())
-				: string.Concat(_cacheKeyPrefix, _fullRegion, ":", key.ToString());
+				? baseKey + "@" + key.GetHashCode()
+				: baseKey;
 
 			if (_maxKeySize < keyAsString.Length)
 			{
-				Log.Info(
-					"Computing a hashed key for too long key '{0}'. This may cause collisions resulting into additional cache misses.",
-					key);
 				// Hash it for respecting max key size. Collisions will be avoided by storing the actual key along
 				// the object and comparing it on retrieval.
-				using (var hasher = new SHA256Managed())
+				var hash = Hasher.HashToString(keyAsString);
+				if (hash.Length > _maxKeySize)
 				{
-					var bytes = Encoding.UTF8.GetBytes(keyAsString);
-					var computedHash = Convert.ToBase64String(hasher.ComputeHash(bytes));
-					if (computedHash.Length <= _maxKeySize)
-						return computedHash;
```

Comment:
Unrelated bug fix by the way. In this case, if the hashed had to be sanitized, it would not have been sanitized.